src/utils.js

```diff
@@ -0,0 +1,257 @@
+    });
+
+    if (!phoneCountryConfig) {
+      return emptyPhoneObject;
+    }
+
+    var carrierCode = matches[3] || '';
+    var number = matches[4];
+
+    return {
+      fkCountry: phoneCountryConfig.iso2Code,
+      cellTokens: {
+        countryCode: countryCode,
+        carrierCode: carrierCode,
+        number: number
+      }
+    };
+  },
+
+  /**
+   * Given an iso2code for a country it will try to find it in the
+   * phoneSettings object that defines the countries you want to accept.
+   *
+   * It will return an object with 2 property:
+   * {
+   *   empty: true/false,
+   *   invalid: true/false,
+   * }
+   *
+   * @param  {String} countryId
+   * @param  {Object} phoneSettings
+   * @return {Object}
+   */
+  validateCountry: function(countryId, phoneSettings) {
+    if (!countryId) {
+      return {empty: true, invalid: true};
+    }
+
+    return {
+      empty: false,
+      invalid: (!phoneSettings[countryId.toLowerCase()] || !phoneSettings[countryId.toLowerCase()].phoneCodes)
+    };
+  },
+
+  /**
+   * Given an iso2code for a country and a carrier code
+   * it will try to find it in the phoneSettings object that
+   * defines the countries and carriers you want to accept.
+   *
+   * It will return an object with 1 or 2 property:
+   *
+   * if countryId is not defined it returns immediatly an object with just the invalid property
+   *
+   * {
+   *   invalid: true,
+   * }
+   *
+   * otherwise it will try to validate the carrier code and return
+   *
+   * {
+   *   empty: true/false,
+   *   invalid: true/false,
+   * }
+   *
+   * @param  {String} carrierCode
+   * @param  {String} countryId
+   * @param  {Object} phoneSettings
+   * @return {Object}
+   */
+  validateCarrierCode: function(carrierCode, countryId, phoneSettings) {
+    if (!countryId) {
+      return {
+        invalid: true
+      };
+    }
+
+    if (!phoneSettings[countryId.toLowerCase()]) {
+      return {
+        invalid: true
+      };
+    }
+
+    var phoneCodes = phoneSettings[countryId.toLowerCase()].phoneCodes;
+
+    if (phoneCodes === undefined) {
+      return {empty: false, invalid: true};
+    }
+
+    if (!carrierCode) {
+      return {
+        empty: true,
+        invalid: !(phoneCodes.carrierCodes === false)
+      };
+    }
+
+    return {
+      empty: false,
+      invalid: !contains(phoneCodes.carrierCodes, carrierCode)
+    };
+  },
+
+  /**
+   * Checks if a number is numeric and
+   * if it matches the min and max length constraints.
+   *
+   * Will return and object that always contains the following properties:
+   *
+   * {
+   *   empty: true/false,
+   *   invalid: true/false,
+   * }
+   *
+   * the object may contain also the following properties to have
+   * a bit more information on what is invalid in the given number:
+   *
+   * {
+   *   empty: true/false,
+   *   invalid: true/false,
+   *   notNumeric: true/false,
+   *   minlengthInvalid: true/false,
+   *   maxlengthInvalid: true/false,
+   * }
```

Comment:
For carrier code it may be that an empty code is a valid one, so you may have 

- `{empty: true, valid: true}` for a country without the carrier code
- `{empty: true, valid: false}` for a country with the carrier code

Moving it inside the `errors` we should handle the first case so that we do not have something like `{valid: true, errors: ['empty']}` (we already have a [specific handler](https://github.com/namshi/nmPhone/pull/1/files/17ede7a30a16e77dd34cd67676eb1109e3fb03f0#diff-2b4ca49d4bb0a774c4d4c1672d7aa781R154) so yes, it may be the same :) ).

I'll give it a shot and see how that goes.